src/moin/scripts/migration/moin19/import19.py

```diff
@@ -136,7 +136,9 @@ def run(self, data_dir=None):
             if namespace == '':
                 namespace = 'default'
             meta, data = backend.retrieve(namespace, revno)
-            data_in = ''.join(data.readlines())
+            charset = meta['contenttype'].split('charset=')[1]
```

Comment:
this code is copy and pasted at quite some places now...

'text/x.moin.wiki;format=1.9;charset=utf-8' <- there you see that this operation is not even working for all the cases (just imagine format and charset were swapped in position).

Reply:
i'll open a ticket for this.